airflow/jobs/job.py

```diff
@@ -213,9 +218,12 @@ def heartbeat(
                 session.merge(self)
                 self.latest_heartbeat = timezone.utcnow()
                 session.commit()
+                # After committing verify if heartbeat has recovered after crossing health check threshold
```

Comment:
Should we add a test case for this behavior?

Reply:
yes let me check it @Lee-W thanks

Reply:
@Lee-W do you have any recommendation on the test case that I can add? I have tried to read test cases in `test_base_job.py` file but I don't know what part of this changes could be added as test in the file.